core/src/main/java/tc/oc/pgm/score/ScoreModule.java

```diff
@@ -94,25 +95,30 @@ public ScoreModule parse(MapFactory factory, Logger logger, Document doc)
       }
 
       RegionParser regionParser = factory.getRegions();
-      ScoreConfig config = new ScoreConfig();
       ImmutableSet.Builder<ScoreBoxFactory> scoreBoxFactories = ImmutableSet.builder();
-
+      Optional<Integer> scoreLimit = Optional.empty();
+      Optional<Integer> mercyLimit = Optional.empty();
+      Optional<Integer> deathScore = Optional.empty();
+      Optional<Integer> killScore = Optional.empty();
       for (Element scoreEl : scoreElements) {
-        config.scoreLimit = XMLUtils.parseNumber(scoreEl.getChild("limit"), Integer.class, -1);
-        config.mercyLimit = XMLUtils.parseNumber(scoreEl.getChild("mercy"), Integer.class, -1);
+        scoreLimit =
+            Optional.of(XMLUtils.parseNumber(scoreEl.getChild("limit"), Integer.class, -1));
```

Comment:
This is not how optionals work, you're defaulting to -1, which will not be an empty optional.

Additionally you're always overriding the previous value so again you're not making proper use of the optionals, it's not changing anything